src/main/java/org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java

```diff
@@ -0,0 +1,128 @@
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.isf.pregtreattype.dto.PregnantTreatmentTypeDTO;
+import org.isf.pregtreattype.manager.PregnantTreatmentTypeBrowserManager;
+import org.isf.pregtreattype.mapper.PregnantTreatmentTypeMapper;
+import org.isf.pregtreattype.model.PregnantTreatmentType;
+import org.isf.shared.exceptions.OHAPIException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.isf.utils.exception.model.OHSeverityLevel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.annotations.Api;
+
+@RestController
+@Api(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+public class PregnantTreatmentTypeController {
+
+	@Autowired
+	protected PregnantTreatmentTypeBrowserManager pregTreatTypeManager;
+	
+	@Autowired
+	protected PregnantTreatmentTypeMapper mapper;
+
+	private final Logger logger = LoggerFactory.getLogger(PregnantTreatmentTypeController.class);
+
+	public PregnantTreatmentTypeController(PregnantTreatmentTypeBrowserManager pregTreatTypeManager, PregnantTreatmentTypeMapper pregnantTreatmentTypemapper) {
+		this.pregTreatTypeManager = pregTreatTypeManager;
+		this.mapper = pregnantTreatmentTypemapper;
+	}
+
+	/**
+	 * create a new {@link PregnantTreatmentType}
+	 * @param pregnantTreatmentTypeDTO
+	 * @return <code>true</code> if the pregnant treatment type has been stored, <code>false</code> otherwise.
+	 * @throws OHServiceException
+	 */
+	@PostMapping(value = "/pregnanttreatmenttypes", produces = MediaType.APPLICATION_JSON_VALUE)
+	ResponseEntity<String> newPregnantTreatmentType(@RequestBody PregnantTreatmentTypeDTO pregnantTreatmentTypeDTO) throws OHServiceException {
+		String code = pregnantTreatmentTypeDTO.getCode();
+		logger.info("Create pregnant treatment Type " + code);
+		boolean isCreated = pregTreatTypeManager.newPregnantTreatmentType(mapper.map2Model(pregnantTreatmentTypeDTO));
+		PregnantTreatmentType pregTreatTypeCreated = null;
+		List<PregnantTreatmentType> pregTreatTypeFounds = pregTreatTypeManager.getPregnantTreatmentType().stream().filter(pregtreattype -> pregtreattype.getCode().equals(code))
+				.collect(Collectors.toList());
+		if (pregTreatTypeFounds.size() > 0)
+			pregTreatTypeCreated = pregTreatTypeFounds.get(0);
```

Comment:
List could be avoided:
`		PregnantTreatmentType pregTreatTypeCreated = pregTreatTypeManager.getPregnantTreatmentType().stream().filter(pregtreattype -> pregtreattype.getCode().equals(code))
				.findFirst().orElse(null);`

Reply:
ok